### DISCUSSION TITLE
How to get User Permission in User Profile Using SPI

### DISCUSSION THREAD

**Discussion by taru-j-a-i-n** [keycloak/keycloak#29178] [Q&A]
I want to create an SPI to retrieve the User Profile having user details, roles, claims and permissins so I am unable t fetch the permission for thatr user how to do it which class is required and wjhat alll Permutaation n combinitions are needed.
sample output of User profile should have userPermissions and claims as not null and not empty

id":"094073fb-0f5e-419f-a582-d25b8b5b5903","email":"admin@admin.com","firstName":"Admin","lastName":"listener","userRoles":["default-roles-portal-realm","create-client","uma_protection","read-token","view-profile","view-groups","view-consent","view-applications","manage-consent","manage-account-links","delete-account"],**"userPermissions":[],"claims":null**}

**Comment by codespearhead** [comment]
@taru-j-a-i-n Can you close thid discussion in favor of #29187 ?

**Reply by taru-j-a-i-n** [reply]
Also how to populate all the claims using SPI

**Reply by taru-j-a-i-n** [reply]
@codespearhead please provide me the readymade keyclao class for getting the claims

**Reply by codespearhead** [reply]
Close the duplicate discussion and I might take a look.

**Reply by taru-j-a-i-n** [reply]
get me the user claims through SPI and I will cloase all the duplicates
